src/backend/distributed/commands/table.c

```diff
@@ -221,14 +221,22 @@ PostprocessCreateTableStmtPartitionOf(CreateStmt *createStatement, const
 	if (IsCitusTable(parentRelationId))
 	{
 		Oid relationId = RangeVarGetRelid(createStatement->relation, NoLock, missingOk);
-		Var *parentDistributionColumn = DistPartitionKeyOrError(parentRelationId);
-		char parentDistributionMethod = DISTRIBUTE_BY_HASH;
-		char *parentRelationName = generate_qualified_relation_name(parentRelationId);
-		bool viaDeprecatedAPI = false;
-
-		CreateDistributedTable(relationId, parentDistributionColumn,
-							   parentDistributionMethod, ShardCount,
-							   parentRelationName, viaDeprecatedAPI);
+
+		/*
+		 * If that partition is already distributed, or if the table is not a partition
+		 * of the current parent, just skip like in plain PG.
+		 */
+		if (!(IsCitusTable(relationId)) && PartitionTable(relationId) &&
```

Comment:
Can you remove the brackets around the `IsCitusTable`. They made me think `!` will cover more checks.